File: BOJ/1316. 그룹 단어 체커/solution.cc
// https://www.acmicpc.net/problem/1316
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int len, cnt = 0;
    cin >> len;

    for (int i = 0; i < len; i += 1) {
        vector<int> a(26, 0);
        int prev;
        string s;
        cin >> s;

        cnt += 1;
        for (int j = 0; j < s.size(); j += 1) {
            int idx = s[j] - 97;
            if (a[idx] == 0) {
                a[idx] = 1;
                prev = idx;
                continue;
            }
            else if (prev != idx && a[idx] > 0) {
                cnt > 0 ? cnt -= 1 : cnt;
                break;
            }
        }
    }
    cout << cnt;
    return 0;
}

File: BOJ/1316. 그룹 단어 체커/solution.js
// https://www.acmicpc.net/problem/1316
const query = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
let cnt = 0;

for (let i = 1; i < query.length; i += 1) {
    const a = Array(26).fill(0);
    let prev;
    cnt += 1;

    for (let j = 0; j < query[i].length; j += 1) {
        const idx = query[i][j].charCodeAt(0) - 97;
        if (a[idx] === 0) {
            a[idx] = 1;
            prev = idx;
            continue;
        }
        else if (prev !== idx && a[idx] > 0) {
            cnt > 0 ? cnt -= 1 : cnt;
            break;
        }
    }
}

console.log(cnt);